auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackIDPPrompt.java

```diff
@@ -186,7 +194,33 @@ public void onComplete(@NonNull Task<AuthResult> task) {
             authResultTask
                     .addOnFailureListener(
                             new TaskFailureLogger(TAG, "Error linking with credential"))
-                    .addOnCompleteListener(new FinishListener());
+                    .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            FirebaseAuth auth = FirebaseAuth.getInstance();
+                            FirebaseUser user = auth.getCurrentUser();
+                            if (!task.isSuccessful() && user != null && mPrevCredential != null) {
+                                user.delete();
+                                final String prevUid = user.getUid();
+                                auth.signInWithCredential(mPrevCredential)
+                                        .addOnFailureListener(
+                                                new TaskFailureLogger(
+                                                        TAG, "Error linking with credential"))
+                                        .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                                            @Override
+                                            public void onComplete(@NonNull Task<AuthResult> task) {
+                                                if (task.isSuccessful()) {
+                                                    AuthUI.getInstance()
+                                                            .notifyOnMergeFailedListeners(prevUid);
```

Comment:
In this block you are notifying the merge failed listeners on successful sign in, why is that the case?